src/WebJobs.Script.WebHost/WebScriptHostHttpRoutesManager.cs

```diff
@@ -87,6 +96,17 @@ public void InitializeHttpFunctionRoutes(IScriptJobHost host)
 
             _router.AddFunctionRoutes(functionRouter, proxyRouter);
 
+            if (warmupRouteBuilder != null)
+            {
+                // Adding the default admin/warmup route when no warmup function is present
+                if (warmupRouteBuilder.Count == 0)
+                {
+                    warmupRouteBuilder.MapFunctionRoute(string.Empty, "warmup", string.Empty);
+                }
+                IRouter warmupRouter = warmupRouteBuilder.Build();
+                _router.AddCustomRoutes(warmupRouter);
```

Comment:
As mentioned in my comment in PR https://github.com/Azure/azure-webjobs-sdk-extensions/pull/668, I'm concerned that this is a breaking change to the public IWebJobsRouter interface

Reply:
Isn't this ok to update the interface under a new version? here is another PR which we updated this interface in the past.
https://github.com/Azure/azure-webjobs-sdk-extensions/pull/443

 I am not sure if there is any cleaner way to add these warmup routes to the routes collection and not have them get mixed up with external http trigger and proxies routes. please take a look at the tests, we should not mix these admin/warmup calls with api/warmup http trigger or proxies if customers happen to have them or with any warmup function which is not a warmup trigger function.

Reply:
It's safe to break in a new major version. That commit you link to was 2 years ago when the extension was still in beta I believe and didn't have its own nuget package, so we hadn't yet released the 3.0 major version with it. These Warmup functions are actual functions, implemented by the user, so why can't we add these via the existing AddFunctionRoutes API along with the other functions? Why doesn't that work?

Reply:
I see, yes makes sense now. sent a PR to revert changes to the interface. will bring that here then.